lib/folio/holdings.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+module Folio
+  class Holdings
+    YEAR_REGEX = /(?:19|20)\d{2}/
+    # We've seen chronologies that looks like full dates, MON YYYY, WIN YYYY, and nil
+    def self.find_latest(holdings)
+      # NOTE: We saw some piece records without 'chronology'. Was this just test data?
+      pieces = holdings.filter_map do |piece|
+        piece.merge(sortable_date: extract_sortable_date(piece['chronology']) || extract_sortable_date(piece['enumeration']),
+                    sortable_tokens: Array(extract_sortable_tokens(piece['chronology']) || extract_sortable_tokens(piece['enumeration']) || piece['enumeration']))
+      end
+      pieces.max_by { |piece| [piece.fetch(:sortable_date)] + piece.fetch(:sortable_tokens) }
+    end
+
+    # Find any dates that are present in the fields
+    def self.extract_sortable_date(date_str)
```

Comment:
I think it's possible we'd have some pieces with a sortable date and some without:

```
> [[Date.parse('2023-01-1'), nil], [nil, nil]].sort
(irb):5:in `sort': comparison of Array with Array failed (ArgumentError)
```

Should this method always return a date object?